generator/recipe.go

```diff
@@ -15,9 +15,13 @@ var file embed.FS
 
 // Template represents the template for generating a recipe.
 type Template struct {
-	Name       string
-	Version    string
-	Source     map[string]string
+	Name    string
+	Version string
+	Source  struct {
+		Type         string
```

Comment:
We can rename `Type` to `Name` since we want to move away from `Type`. wdyt?

Reply:
sure, will do